views/reports/revised_idsr_monthly_menu.html

```diff
@@ -64,26 +64,35 @@
   fetchMonths();
 
 function load(months) {
-  var sel = document.getElementById('months');
-  var opt = document.createElement('option');
-
-  if (document.readyState === "complete" || document.readyState === "interactive") {
-    for (var i=0; i<months.length; i++) {
-    var payload = months[i];
-    var _payload = payload[1];
-    var month_and_year = _payload[0]
-    var start_and_end_months = _payload[1];
-    var opt = document.createElement('option');
-    var dates = start_and_end_months.split("to");
-    var month = month_and_year.split('-')[0];
-    opt.innerHTML = month_and_year;
-    var value = dates[0]+ " " +dates[1] +" "+ month;
-    opt.setAttribute('value', value);
-    sel.appendChild(opt);  
-  }
-  } else {
-    fetchMonths();
-  }
+  let promise = new Promise((resolve, reject) => {
+    var selection = document.getElementById('months')
+
+    if(typeof(selection) != undefined) {
+      for (var i=0; i<months.length; i++) {
+      var payload = months[i];
+      var _payload = payload[1];
+      var month_and_year = _payload[0]
+      var start_and_end_months = _payload[1];
+      var opt = document.createElement('option');
+      var dates = start_and_end_months.split("to");
+      var month = month_and_year.split('-')[0];
+      opt.innerHTML = month_and_year;
+      var value = dates[0]+ " " +dates[1] +" "+ month;
+      opt.setAttribute('value', value);
+      selection.appendChild(opt);  
+    }
+      resolve("success")
+      //hack to promise
+      console.log(selection)
+    } else {
+      reject("failed")
+    }
+  })
+  promise.then((message) => {
+    console.log(message)
+  }).catch((message) => {
+    console.log(message)
+  })
```

Comment:
For the purposes of this application and considering we don't have a build system like babel to transpile this to older versions of JS, we ususally try to refrain from using es6 code in this application considering some sites still use older versions of firefox that may cause some issues. please use xhr functions for now to make the code bulletproof in terms of compatibility. so arrow functions, let, const, and promises should be avoided as much as possible

Reply:
noted we will remove them